{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "lines_to_next_cell": 0
   },
   "source": [
    "# CB2-101: Sequence Similarity Search\n",
    "Malay (malay@uab.edu)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Note\n",
    "\n",
    "NCBI has recently removed the C blast programs from there website. Please download last C BLAST from our own copy to use the example in this handout."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-10-13 20:23:19--  http://cmb.path.uab.edu/training/2018/files/blast-2.2.26.tar.gz\n",
      "Resolving cmb.path.uab.edu (cmb.path.uab.edu)... 138.26.58.200\n",
      "Connecting to cmb.path.uab.edu (cmb.path.uab.edu)|138.26.58.200|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 40490310 (39M) [application/x-gzip]\n",
      "Saving to: ‘blast-2.2.26.tar.gz’\n",
      "\n",
      "100%[======================================>] 40,490,310  86.8MB/s   in 0.4s   \n",
      "\n",
      "2021-10-13 20:23:19 (86.8 MB/s) - ‘blast-2.2.26.tar.gz’ saved [40490310/40490310]\n",
      "\n",
      "blast-2.2.26/\n",
      "blast-2.2.26/bin/\n",
      "blast-2.2.26/bin/bl2seq\n",
      "blast-2.2.26/bin/blastall\n",
      "blast-2.2.26/bin/blastclust\n",
      "blast-2.2.26/bin/blastpgp\n",
      "blast-2.2.26/bin/copymat\n",
      "blast-2.2.26/bin/fastacmd\n",
      "blast-2.2.26/bin/formatdb\n",
      "blast-2.2.26/bin/formatrpsdb\n",
      "blast-2.2.26/bin/impala\n",
      "blast-2.2.26/bin/makemat\n",
      "blast-2.2.26/bin/megablast\n",
      "blast-2.2.26/bin/rpsblast\n",
      "blast-2.2.26/bin/seedtop\n",
      "blast-2.2.26/data/\n",
      "blast-2.2.26/data/BLOSUM45\n",
      "blast-2.2.26/data/BLOSUM62\n",
      "blast-2.2.26/data/BLOSUM80\n",
      "blast-2.2.26/data/KSat.flt\n",
      "blast-2.2.26/data/KSchoth.flt\n",
      "blast-2.2.26/data/KSgc.flt\n",
      "blast-2.2.26/data/KShopp.flt\n",
      "blast-2.2.26/data/KSkyte.flt\n",
      "blast-2.2.26/data/KSpcc.mat\n",
      "blast-2.2.26/data/KSpur.flt\n",
      "blast-2.2.26/data/KSpyr.flt\n",
      "blast-2.2.26/data/PAM30\n",
      "blast-2.2.26/data/PAM70\n",
      "blast-2.2.26/data/UniVec.nhr\n",
      "blast-2.2.26/data/UniVec.nin\n",
      "blast-2.2.26/data/UniVec.nsq\n",
      "blast-2.2.26/data/UniVec_Core.nhr\n",
      "blast-2.2.26/data/UniVec_Core.nin\n",
      "blast-2.2.26/data/UniVec_Core.nsq\n",
      "blast-2.2.26/data/asn2ff.prt\n",
      "blast-2.2.26/data/bstdt.val\n",
      "blast-2.2.26/data/ecnum_ambiguous.txt\n",
      "blast-2.2.26/data/ecnum_specific.txt\n",
      "blast-2.2.26/data/featdef.val\n",
      "blast-2.2.26/data/gc.val\n",
      "blast-2.2.26/data/humrep.fsa\n",
      "blast-2.2.26/data/lineages.txt\n",
      "blast-2.2.26/data/makerpt.prt\n",
      "blast-2.2.26/data/objprt.prt\n",
      "blast-2.2.26/data/organelle_products.prt\n",
      "blast-2.2.26/data/product_rules.prt\n",
      "blast-2.2.26/data/pubkey.enc\n",
      "blast-2.2.26/data/seqcode.val\n",
      "blast-2.2.26/data/sequin.hlp\n",
      "blast-2.2.26/data/sgmlbb.ent\n",
      "blast-2.2.26/data/taxlist.txt\n",
      "blast-2.2.26/doc/\n",
      "blast-2.2.26/doc/bl2seq.html\n",
      "blast-2.2.26/doc/blast.html\n",
      "blast-2.2.26/doc/blastall.html\n",
      "blast-2.2.26/doc/blastclust.html\n",
      "blast-2.2.26/doc/blastdb.html\n",
      "blast-2.2.26/doc/blastftp.html\n",
      "blast-2.2.26/doc/blastpgp.html\n",
      "blast-2.2.26/doc/fastacmd.html\n",
      "blast-2.2.26/doc/filter.html\n",
      "blast-2.2.26/doc/formatdb.html\n",
      "blast-2.2.26/doc/formatrpsdb.html\n",
      "blast-2.2.26/doc/history.html\n",
      "blast-2.2.26/doc/impala.html\n",
      "blast-2.2.26/doc/index.html\n",
      "blast-2.2.26/doc/megablast.html\n",
      "blast-2.2.26/doc/netblast.html\n",
      "blast-2.2.26/doc/rpsblast.html\n",
      "blast-2.2.26/doc/scoring.pdf\n",
      "blast-2.2.26/doc/web_blast.pl\n",
      "blast-2.2.26/VERSION\n"
     ]
    }
   ],
   "source": [
    "wget \"http://cmb.path.uab.edu/training/2018/files/blast-2.2.26.tar.gz\"\n",
    "tar -xvzf blast-2.2.26.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Put blast in path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd blast-2.2.26/bin\n",
    "export PATH=$PATH:`pwd`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that `blast` is in your path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/user/home/secook2/info302/info302_assignment/blast-2.2.26/bin/blastall\n"
     ]
    }
   ],
   "source": [
    "which blastall\n",
    "cd ../.."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that you are in outside of blast directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/user/home/secook2/info302/info302_assignment\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simple BLAST\n",
    "\n",
    "BLAST p53 human sequence against SwissProt database.\n",
    "\n",
    "## Download sequence files\n",
    "\n",
    "Download SwissProt FASTA file file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-10-13 20:27:17--  ftp://ftp.expasy.org/databases/swiss-prot/release_compressed/uniprot_sprot.fasta.gz\n",
      "           => ‘uniprot_sprot.fasta.gz’\n",
      "Resolving ftp.expasy.org (ftp.expasy.org)... 192.33.215.162\n",
      "Connecting to ftp.expasy.org (ftp.expasy.org)|192.33.215.162|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /databases/swiss-prot/release_compressed ... done.\n",
      "==> SIZE uniprot_sprot.fasta.gz ... 90275536\n",
      "==> PASV ... done.    ==> RETR uniprot_sprot.fasta.gz ... done.\n",
      "Length: 90275536 (86M) (unauthoritative)\n",
      "\n",
      "100%[======================================>] 90,275,536  20.8MB/s   in 8.4s   \n",
      "\n",
      "2021-10-13 20:27:26 (10.2 MB/s) - ‘uniprot_sprot.fasta.gz’ saved [90275536]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget  ftp://ftp.expasy.org/databases/swiss-prot/release_compressed/uniprot_sprot.fasta.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find the id of P53 using the Uniprot API."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "wget -q -O p53.fas http://www.uniprot.org/uniprot/P53_HUMAN.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Create the BLAST datbase and run BLASTP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check the `formatdb` command help. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "formatdb 2.2.26   arguments:\n",
      "\n",
      "  -t  Title for database file [String]  Optional\n",
      "  -i  Input file(s) for formatting [File In]  Optional\n",
      "  -l  Logfile name: [File Out]  Optional\n",
      "    default = formatdb.log\n",
      "  -p  Type of file\n",
      "         T - protein   \n",
      "         F - nucleotide [T/F]  Optional\n",
      "    default = T\n",
      "  -o  Parse options\n",
      "         T - True: Parse SeqId and create indexes.\n",
      "         F - False: Do not parse SeqId. Do not create indexes.\n",
      " [T/F]  Optional\n",
      "    default = F\n",
      "  -a  Input file is database in ASN.1 format (otherwise FASTA is expected)\n",
      "         T - True, \n",
      "         F - False.\n",
      " [T/F]  Optional\n",
      "    default = F\n",
      "  -b  ASN.1 database in binary mode\n",
      "         T - binary, \n",
      "         F - text mode.\n",
      " [T/F]  Optional\n",
      "    default = F\n",
      "  -e  Input is a Seq-entry [T/F]  Optional\n",
      "    default = F\n",
      "  -n  Base name for BLAST files [String]  Optional\n",
      "  -v  Database volume size in millions of letters [Integer]  Optional\n",
      "    default = 4000\n",
      "  -s  Create indexes limited only to accessions - sparse [T/F]  Optional\n",
      "    default = F\n",
      "  -V  Verbose: check for non-unique string ids in the database [T/F]  Optional\n",
      "    default = F\n",
      "  -L  Create an alias file with this name\n",
      "        use the gifile arg (below) if set to calculate db size\n",
      "        use the BLAST db specified with -i (above) [File Out]  Optional\n",
      "  -F  Gifile (file containing list of gi's) [File In]  Optional\n",
      "  -B  Binary Gifile produced from the Gifile specified above [File Out]  Optional\n",
      "  -T  Taxid file to set the taxonomy ids in ASN.1 deflines [File In]  Optional\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "formatdb -"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "engine": "bash",
    "eval": false
   },
   "outputs": [],
   "source": [
    "zcat uniprot_sprot.fasta.gz | formatdb -i stdin -t \"swissprot\" -o T -n \"swissprot\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note how we used the compressed file in a pipe to create the blast database. Now we will run the actual blast to search the p53 gene in Swissprot database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "blastall -p blastp -i p53.fas -d swissprot -o output.bla"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get a different output format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "blastall -p blastp -i p53.fas -d swissprot -o output_tabular.bla -m 9"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Is P53 is present in yeast genome?\n",
    "\n",
    "In this excerise you will download yeast genome and search the same p53 fasta agains yeast genome. The idea is to see whether p53 is present in yeast genome.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "First let's create the SC proteome file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "cache": true,
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 2
   },
   "outputs": [],
   "source": [
    "# Download all the files of SC from NCBI\n",
    "wget --quiet -O sc.faa.gz ftp://ftp.ncbi.nih.gov/genomes/refseq/fungi/Saccharomyces_cerevisiae/reference/GCF_000146045.2_R64/GCF_000146045.2_R64_protein.faa.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now format the database as shown above and run the p53.fas file against this database. What is your observation?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "cache": true,
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "zcat uniprot_sprot.fasta.gz | formatdb -i stdin -t \"yeast\" -o T -n \"yeast\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "lines_to_next_cell": 0
   },
   "source": [
    "\n",
    "\n",
    "## HMMER\n",
    "\n",
    "```HMMER``` is the only known software for HMM use in bioinformatics. You can download HMMER from \n",
    "\n",
    "http://hmmer.janelia.org/software\n",
    "\n",
    "There are quite a few software that are bundled with HMMER distribution. But the 4 most common ones are:\n",
    "\n",
    "1. ```hmmsearch``` - Searches HMMs against protein sequences\n",
    "2. ```hmmscan``` - Searches protein sequences against HMM library\n",
    "3. ```hmmbuild``` - Builds HMM from a multiple alignment\n",
    "4. ```hmmpress``` - Convert a flat file HMM to binary format that can be used with the software\n",
    "\n",
    "\n",
    "We will search the SwissProt data with a profile of P53 gene.  \n",
    "\n",
    "We will first get a bunch of orthologs of P53 from the Homologene database.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "wget --quiet ftp://ftp.ncbi.nih.gov/pub/HomoloGene/current/homologene.data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will grep the file to find p53."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "engine": "bash",
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "cat homologene.data | grep -i tp53 | head -n 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "lines_to_next_cell": 0
   },
   "source": [
    "Looks like cluster 460 contains T53. We will use a bit of shell script to get the accession."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NP_000537.3\n",
      "XP_001172077.2\n",
      "NP_001040616.1\n",
      "NP_001003210.1\n",
      "NP_776626.1\n",
      "NP_112251.2\n",
      "NP_001258749.1\n",
      "NP_001001903.1\n"
     ]
    }
   ],
   "source": [
    "cat homologene.data | grep \"^460\" | grep -w -i TP53 | cut -f 6 >p53_homologene_ids.txt\n",
    "cat p53_homologene_ids.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will now use NCBI eutils to extract those sesquence. Let's create a fasta file with this ids:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [],
   "source": [
    "for i in `cat p53_homologene_ids.txt`\n",
    "do \n",
    "wget -q -O - \"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?\\\n",
    "db=protein&id=$i&rettype=fasta&retmode=text\"\n",
    "done >p53_all.fas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will first use ```muscle``` to align those sequence and create the HMM, then search SC genome with it. First and put muscle in path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-10-13 20:52:46--  https://www.drive5.com/muscle/downloads3.8.31/muscle3.8.31_i86linux64.tar.gz\n",
      "Resolving www.drive5.com (www.drive5.com)... 199.195.116.69\n",
      "Connecting to www.drive5.com (www.drive5.com)|199.195.116.69|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 487906 (476K) [application/x-gzip]\n",
      "Saving to: ‘muscle3.8.31_i86linux64.tar.gz’\n",
      "\n",
      "100%[======================================>] 487,906     --.-K/s   in 0.1s    \n",
      "\n",
      "2021-10-13 20:52:46 (3.65 MB/s) - ‘muscle3.8.31_i86linux64.tar.gz’ saved [487906/487906]\n",
      "\n",
      "muscle3.8.31_i86linux64\n",
      "/data/user/home/secook2/info302/info302_assignment/blast-2.2.26/bin/muscle\n"
     ]
    }
   ],
   "source": [
    "wget https://www.drive5.com/muscle/downloads3.8.31/muscle3.8.31_i86linux64.tar.gz\n",
    "tar -xvzf muscle3.8.31_i86linux64.tar.gz\n",
    "mv muscle3.8.31_i86linux64 muscle\n",
    "which muscle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download and prepare `hmmer`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-10-13 20:56:18--  http://eddylab.org/software/hmmer/hmmer-3.3.tar.gz\n",
      "Resolving eddylab.org (eddylab.org)... 96.126.110.11, 2600:3c03::f03c:91ff:fec8:383c\n",
      "Connecting to eddylab.org (eddylab.org)|96.126.110.11|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 18614771 (18M) [application/x-gzip]\n",
      "Saving to: ‘hmmer-3.3.tar.gz’\n",
      "\n",
      "100%[======================================>] 18,614,771  14.0MB/s   in 1.3s   \n",
      "\n",
      "2021-10-13 20:56:19 (14.0 MB/s) - ‘hmmer-3.3.tar.gz’ saved [18614771/18614771]\n",
      "\n",
      "hmmer-3.3/\n",
      "hmmer-3.3/documentation/\n",
      "hmmer-3.3/install-sh\n",
      "hmmer-3.3/configure.ac\n",
      "hmmer-3.3/tutorial/\n",
      "hmmer-3.3/INSTALL\n",
      "hmmer-3.3/libdivsufsort/\n",
      "hmmer-3.3/LICENSE\n",
      "hmmer-3.3/test-speed/\n",
      "hmmer-3.3/configure\n",
      "hmmer-3.3/testsuite/\n",
      "hmmer-3.3/config.guess\n",
      "hmmer-3.3/README.md\n",
      "hmmer-3.3/Userguide.pdf\n",
      "hmmer-3.3/config.sub\n",
      "hmmer-3.3/RELEASE-3.3.md\n",
      "hmmer-3.3/.gitignore\n",
      "hmmer-3.3/profmark/\n",
      "hmmer-3.3/makeTAGS.sh\n",
      "hmmer-3.3/easel/\n",
      "hmmer-3.3/Makefile.in\n",
      "hmmer-3.3/.travis.yml\n",
      "hmmer-3.3/src/\n",
      "hmmer-3.3/src/itest_brute.c\n",
      "hmmer-3.3/src/p7_hmmd_search_stats.c\n",
      "hmmer-3.3/src/p7_gmx.c\n",
      "hmmer-3.3/src/hmmdutils.c\n",
      "hmmer-3.3/src/hmmd_search_status.c\n",
      "hmmer-3.3/src/hmmpgmd_client_example.pl\n",
      "hmmer-3.3/src/hmmdwrkr_shard.c\n",
      "hmmer-3.3/src/cachedb_shard.h\n",
      "hmmer-3.3/src/p7_tophits.c\n",
      "hmmer-3.3/src/generic_fwdback_banded.c\n",
      "hmmer-3.3/src/hmmscan.c\n",
      "hmmer-3.3/src/modelconfig.c\n",
      "hmmer-3.3/src/p7_gmxchk.h\n",
      "hmmer-3.3/src/generic_stotrace.c\n",
      "hmmer-3.3/src/generic_null2.c\n",
      "hmmer-3.3/src/p7_alidisplay.c\n",
      "hmmer-3.3/src/impl_sse/\n",
      "hmmer-3.3/src/evalues.c\n",
      "hmmer-3.3/src/hmmemit.c\n",
      "hmmer-3.3/src/p7_gbands.c\n",
      "hmmer-3.3/src/p7_hmmcache.h\n",
      "hmmer-3.3/src/emit.c\n",
      "hmmer-3.3/src/p7_gmxb.c\n",
      "hmmer-3.3/src/hmmpress.c\n",
      "hmmer-3.3/src/hmmbuild.c\n",
      "hmmer-3.3/src/cachedb.h\n",
      "hmmer-3.3/src/alimask.c\n",
      "hmmer-3.3/src/nhmmscan.c\n",
      "hmmer-3.3/src/hmmpgmd.c\n",
      "hmmer-3.3/src/p7_prior.c\n",
      "hmmer-3.3/src/p7_scoredata.c\n",
      "hmmer-3.3/src/generic_fwdback_chk.c\n",
      "hmmer-3.3/src/impl_vmx/\n",
      "hmmer-3.3/src/tracealign.c\n",
      "hmmer-3.3/src/hmmpress.itest.pl\n",
      "hmmer-3.3/src/p7_builder.c\n",
      "hmmer-3.3/src/p7_bg.c\n",
      "hmmer-3.3/src/hmmpgmd_shard.h\n",
      "hmmer-3.3/src/hmmer.h\n",
      "hmmer-3.3/src/p7_hit.c\n",
      "hmmer-3.3/src/seqmodel.c\n",
      "hmmer-3.3/src/generic_decoding.c\n",
      "hmmer-3.3/src/hmmalign.c\n",
      "hmmer-3.3/src/fm_alphabet.c\n",
      "hmmer-3.3/src/generic_viterbi.c\n",
      "hmmer-3.3/src/h2_io.c\n",
      "hmmer-3.3/src/fm_sse.c\n",
      "hmmer-3.3/src/p7_domaindef.c\n",
      "hmmer-3.3/src/generic_fwdback_rescaled.c\n",
      "hmmer-3.3/src/p7_spensemble.c\n",
      "hmmer-3.3/src/cachedb_shard.c\n",
      "hmmer-3.3/src/hmmdmstr.c\n",
      "hmmer-3.3/src/p7_hmmwindow.c\n",
      "hmmer-3.3/src/p7_hmm.c\n",
      "hmmer-3.3/src/nhmmer.c\n",
      "hmmer-3.3/src/fm_general.c\n",
      "hmmer-3.3/src/hmmdwrkr.c\n",
      "hmmer-3.3/src/generic_msv.c\n",
      "hmmer-3.3/src/hmmsim.c\n",
      "hmmer-3.3/src/fm_ssv.c\n",
      "hmmer-3.3/src/hmmlogo.c\n",
      "hmmer-3.3/src/build.c\n",
      "hmmer-3.3/src/makehmmerdb.c\n",
      "hmmer-3.3/src/jackhmmer.c\n",
      "hmmer-3.3/src/p7_pipeline.c\n",
      "hmmer-3.3/src/generic_optacc.c\n",
      "hmmer-3.3/src/modelstats.c\n",
      "hmmer-3.3/src/p7_config.h.in\n",
      "hmmer-3.3/src/hmmc2.c\n",
      "hmmer-3.3/src/heatmap.c\n",
      "hmmer-3.3/src/logsum.c\n",
      "hmmer-3.3/src/hmmstat.c\n",
      "hmmer-3.3/src/generic_fwdback.c\n",
      "hmmer-3.3/src/p7_gmxchk.c\n",
      "hmmer-3.3/src/phmmer.c\n",
      "hmmer-3.3/src/hmmsearch.c\n",
      "hmmer-3.3/src/generic_vtrace.c\n",
      "hmmer-3.3/src/mpisupport.c\n",
      "hmmer-3.3/src/cachedb.c\n",
      "hmmer-3.3/src/hmmpgmd.h\n",
      "hmmer-3.3/src/p7_gmxb.h\n",
      "hmmer-3.3/src/p7_null3.c\n",
      "hmmer-3.3/src/p7_trace.c\n",
      "hmmer-3.3/src/p7_hmmcache.c\n",
      "hmmer-3.3/src/p7_gbands.h\n",
      "hmmer-3.3/src/hmmerfm-exactmatch.c\n",
      "hmmer-3.3/src/Makefile.in\n",
      "hmmer-3.3/src/eweight.c\n",
      "hmmer-3.3/src/hmmpgmd2msa.c\n",
      "hmmer-3.3/src/hmmer.c\n",
      "hmmer-3.3/src/hmmpgmd_shard.c\n",
      "hmmer-3.3/src/p7_profile.c\n",
      "hmmer-3.3/src/hmmdmstr_shard.c\n",
      "hmmer-3.3/src/hmmconvert.c\n",
      "hmmer-3.3/src/p7_domain.c\n",
      "hmmer-3.3/src/errors.c\n",
      "hmmer-3.3/src/hmmfetch.c\n",
      "hmmer-3.3/src/p7_hmmfile.c\n",
      "hmmer-3.3/src/impl_vmx/p7_oprofile.c\n",
      "hmmer-3.3/src/impl_vmx/optacc.c\n",
      "hmmer-3.3/src/impl_vmx/vitscore.c\n",
      "hmmer-3.3/src/impl_vmx/io.c\n",
      "hmmer-3.3/src/impl_vmx/decoding.c\n",
      "hmmer-3.3/src/impl_vmx/msvfilter.c\n",
      "hmmer-3.3/src/impl_vmx/impl_vmx.h\n",
      "hmmer-3.3/src/impl_vmx/fwdback.c\n",
      "hmmer-3.3/src/impl_vmx/p7_omx.c\n",
      "hmmer-3.3/src/impl_vmx/null2.c\n",
      "hmmer-3.3/src/impl_vmx/vitfilter.c\n",
      "hmmer-3.3/src/impl_vmx/stotrace.c\n",
      "hmmer-3.3/src/impl_vmx/00MANIFEST\n",
      "hmmer-3.3/src/impl_vmx/Makefile.in\n",
      "hmmer-3.3/src/impl_vmx/mpi.c\n",
      "hmmer-3.3/src/impl_sse/p7_oprofile.c\n",
      "hmmer-3.3/src/impl_sse/optacc.c\n",
      "hmmer-3.3/src/impl_sse/vitscore.c\n",
      "hmmer-3.3/src/impl_sse/io.c\n",
      "hmmer-3.3/src/impl_sse/decoding.c\n",
      "hmmer-3.3/src/impl_sse/msvfilter.c\n",
      "hmmer-3.3/src/impl_sse/fbparsers.tex\n",
      "hmmer-3.3/src/impl_sse/ssvfilter.c\n",
      "hmmer-3.3/src/impl_sse/impl_sse.h\n",
      "hmmer-3.3/src/impl_sse/fwdback.c\n",
      "hmmer-3.3/src/impl_sse/p7_omx.c\n",
      "hmmer-3.3/src/impl_sse/null2.c\n",
      "hmmer-3.3/src/impl_sse/p7_omx.ai\n",
      "hmmer-3.3/src/impl_sse/vitfilter.c\n",
      "hmmer-3.3/src/impl_sse/stotrace.c\n",
      "hmmer-3.3/src/impl_sse/00MANIFEST\n",
      "hmmer-3.3/src/impl_sse/Makefile.in\n",
      "hmmer-3.3/src/impl_sse/mpi.c\n",
      "hmmer-3.3/easel/esl_sse.h\n",
      "hmmer-3.3/easel/esl_regexp.tex\n",
      "hmmer-3.3/easel/esl_exponential.c\n",
      "hmmer-3.3/easel/esl_msashuffle.c\n",
      "hmmer-3.3/easel/esl_alphabet.h\n",
      "hmmer-3.3/easel/esl_random.h\n",
      "hmmer-3.3/easel/esl_msafile_stockholm.c\n",
      "hmmer-3.3/easel/esl_msafile_a2m.c\n",
      "hmmer-3.3/easel/esl_recorder.h\n",
      "hmmer-3.3/easel/esl_gev.tex\n",
      "hmmer-3.3/easel/esl_vmx.h\n",
      "hmmer-3.3/easel/esl_cluster.h\n",
      "hmmer-3.3/easel/miniapps/\n",
      "hmmer-3.3/easel/esl_ssi.c\n",
      "hmmer-3.3/easel/esl_heap.h\n",
      "hmmer-3.3/easel/esl_hyperexp.tex\n",
      "hmmer-3.3/easel/esl_mpi.tex\n",
      "hmmer-3.3/easel/esl_wuss.tex\n",
      "hmmer-3.3/easel/esl_subcmd.c\n",
      "hmmer-3.3/easel/esl_dmatrix.tex\n",
      "hmmer-3.3/easel/esl_keyhash.h\n",
      "hmmer-3.3/easel/esl_arr3.h\n",
      "hmmer-3.3/easel/esl_sq.c\n",
      "hmmer-3.3/easel/esl_mem.h\n",
      "hmmer-3.3/easel/esl_stretchexp.h\n",
      "hmmer-3.3/easel/esl_matrixops.c\n",
      "hmmer-3.3/easel/documentation/\n",
      "hmmer-3.3/easel/esl_getopts.c\n",
      "hmmer-3.3/easel/esl_dsqdata.tex\n",
      "hmmer-3.3/easel/install-sh\n",
      "hmmer-3.3/easel/configure.ac\n",
      "hmmer-3.3/easel/esl_regexp.h\n",
      "hmmer-3.3/easel/esl_getopts.md\n",
      "hmmer-3.3/easel/esl_dirichlet.h\n",
      "hmmer-3.3/easel/esl_sqio.h\n",
      "hmmer-3.3/easel/esl_randomseq.c\n",
      "hmmer-3.3/easel/esl_randomseq.tex\n",
      "hmmer-3.3/easel/esl_stopwatch.tex\n",
      "hmmer-3.3/easel/esl_quicksort.h\n",
      "hmmer-3.3/easel/esl_json.h\n",
      "hmmer-3.3/easel/esl_fileparser.h\n",
      "hmmer-3.3/easel/interface_gsl.h\n",
      "hmmer-3.3/easel/formats/\n",
      "hmmer-3.3/easel/esl_hmm.c\n",
      "hmmer-3.3/easel/esl_swat.tex\n",
      "hmmer-3.3/easel/esl_stopwatch.c\n",
      "hmmer-3.3/easel/esl_neon.c\n",
      "hmmer-3.3/easel/esl_rand64.h\n",
      "hmmer-3.3/easel/esl_msa.tex\n",
      "hmmer-3.3/easel/esl_mpi.c\n",
      "hmmer-3.3/easel/LICENSE\n",
      "hmmer-3.3/easel/esl_gencode.c\n",
      "hmmer-3.3/easel/esl_stats.c\n",
      "hmmer-3.3/easel/esl_red_black.h\n",
      "hmmer-3.3/easel/demotic/\n",
      "hmmer-3.3/easel/esl_scorematrix.h\n",
      "hmmer-3.3/easel/esl_avx512.c\n",
      "hmmer-3.3/easel/esl_alloc.c\n",
      "hmmer-3.3/easel/easel.c\n",
      "hmmer-3.3/easel/esl_graph.h\n",
      "hmmer-3.3/easel/esl_distance.c\n",
      "hmmer-3.3/easel/esl_histogram.c\n",
      "hmmer-3.3/easel/esl_composition.h\n",
      "hmmer-3.3/easel/esl_gev.c\n",
      "hmmer-3.3/easel/esl_tree.tex\n",
      "hmmer-3.3/easel/esl_weibull.h\n",
      "hmmer-3.3/easel/esl_paml.c\n",
      "hmmer-3.3/easel/esl_msafile_phylip.c\n",
      "hmmer-3.3/easel/esl_workqueue.h\n",
      "hmmer-3.3/easel/esl_msa.c\n",
      "hmmer-3.3/easel/esl_msafile2.h\n",
      "hmmer-3.3/easel/esl_tree.c\n",
      "hmmer-3.3/easel/esl_buffer.tex\n",
      "hmmer-3.3/easel/esl_sqio_ascii.h\n",
      "hmmer-3.3/easel/esl_hyperexp.h\n",
      "hmmer-3.3/easel/esl_arr2.c\n",
      "hmmer-3.3/easel/esl_weibull.tex\n",
      "hmmer-3.3/easel/esl_vectorops.c\n",
      "hmmer-3.3/easel/testsuite/\n",
      "hmmer-3.3/easel/esl_ratematrix.c\n",
      "hmmer-3.3/easel/esl_msaweight.c\n",
      "hmmer-3.3/easel/esl_rootfinder.h\n",
      "hmmer-3.3/easel/esl_sqio_ncbi.c\n",
      "hmmer-3.3/easel/esl_normal.tex\n",
      "hmmer-3.3/easel/esl_mixgev.c\n",
      "hmmer-3.3/easel/esl_alphabet.tex\n",
      "hmmer-3.3/easel/esl_dsqdata.md\n",
      "hmmer-3.3/easel/esl_msafile_psiblast.c\n",
      "hmmer-3.3/easel/esl_bitfield.h\n",
      "hmmer-3.3/easel/esl_msafile_selex.h\n",
      "hmmer-3.3/easel/esl_msacluster.h\n",
      "hmmer-3.3/easel/esl_threads.h\n",
      "hmmer-3.3/easel/esl_minimizer.c\n",
      "hmmer-3.3/easel/esl_dsqdata.c\n",
      "hmmer-3.3/easel/esl_varint.c\n",
      "hmmer-3.3/easel/esl_swat.c\n",
      "hmmer-3.3/easel/esl_normal.h\n",
      "hmmer-3.3/easel/esl_cpu.h\n",
      "hmmer-3.3/easel/interface_lapack.h\n",
      "hmmer-3.3/easel/esl_vectorops.md\n",
      "hmmer-3.3/easel/esl_wuss.h\n",
      "hmmer-3.3/easel/esl_gamma.c\n",
      "hmmer-3.3/easel/esl_cluster.tex\n",
      "hmmer-3.3/easel/config.guess\n",
      "hmmer-3.3/easel/esl_gumbel.h\n",
      "hmmer-3.3/easel/esl_msafile_clustal.c\n",
      "hmmer-3.3/easel/esl_dmatrix.h\n",
      "hmmer-3.3/easel/esl_mixdchlet.h\n",
      "hmmer-3.3/easel/esl_sqio.tex\n",
      "hmmer-3.3/easel/esl_huffman.c\n",
      "hmmer-3.3/easel/esl_stack.c\n",
      "hmmer-3.3/easel/esl_msafile.h\n",
      "hmmer-3.3/easel/esl_msafile_afa.c\n",
      "hmmer-3.3/easel/esl_histogram.tex\n",
      "hmmer-3.3/easel/esl_config.h.in\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hmmer-3.3/easel/release-notes/\n",
      "hmmer-3.3/easel/esl_buffer.h\n",
      "hmmer-3.3/easel/esl_scorematrix.tex\n",
      "hmmer-3.3/easel/esl_random.tex\n",
      "hmmer-3.3/easel/esl_varint.md\n",
      "hmmer-3.3/easel/esl_avx.c\n",
      "hmmer-3.3/easel/esl_sse.tex\n",
      "hmmer-3.3/easel/esl_mem.md\n",
      "hmmer-3.3/easel/README.md\n",
      "hmmer-3.3/easel/esl_dirichlet.c\n",
      "hmmer-3.3/easel/esl_regexp.c\n",
      "hmmer-3.3/easel/esl_stretchexp.c\n",
      "hmmer-3.3/easel/esl_sq.h\n",
      "hmmer-3.3/easel/esl_mem.c\n",
      "hmmer-3.3/easel/esl_keyhash.c\n",
      "hmmer-3.3/easel/esl_arr3.c\n",
      "hmmer-3.3/easel/esl_subcmd.h\n",
      "hmmer-3.3/easel/esl_getopts.h\n",
      "hmmer-3.3/easel/esl_matrixops.h\n",
      "hmmer-3.3/easel/esl_ssi.tex\n",
      "hmmer-3.3/easel/config.sub\n",
      "hmmer-3.3/easel/esl_keyhash.tex\n",
      "hmmer-3.3/easel/esl_vmx.c\n",
      "hmmer-3.3/easel/esl_recorder.c\n",
      "hmmer-3.3/easel/esl_msafile_stockholm.h\n",
      "hmmer-3.3/easel/esl_msafile_a2m.h\n",
      "hmmer-3.3/easel/esl_gamma.tex\n",
      "hmmer-3.3/easel/esl_heap.c\n",
      "hmmer-3.3/easel/esl_ssi.h\n",
      "hmmer-3.3/easel/esl_cluster.c\n",
      "hmmer-3.3/easel/esl_sse.c\n",
      "hmmer-3.3/easel/.gitignore\n",
      "hmmer-3.3/easel/esl_random.c\n",
      "hmmer-3.3/easel/esl_alphabet.c\n",
      "hmmer-3.3/easel/esl_msashuffle.h\n",
      "hmmer-3.3/easel/esl_exponential.h\n",
      "hmmer-3.3/easel/esl_graph.c\n",
      "hmmer-3.3/easel/esl_distance.h\n",
      "hmmer-3.3/easel/esl_paml.h\n",
      "hmmer-3.3/easel/esl_weibull.c\n",
      "hmmer-3.3/easel/esl_distance.tex\n",
      "hmmer-3.3/easel/esl_gev.h\n",
      "hmmer-3.3/easel/esl_gumbel.tex\n",
      "hmmer-3.3/easel/esl_composition.c\n",
      "hmmer-3.3/easel/esl_histogram.h\n",
      "hmmer-3.3/easel/esl_red_black.c\n",
      "hmmer-3.3/easel/esl_scorematrix.c\n",
      "hmmer-3.3/easel/esl_avx512.h\n",
      "hmmer-3.3/easel/esl_ratematrix.tex\n",
      "hmmer-3.3/easel/esl_stats.h\n",
      "hmmer-3.3/easel/esl_gencode.h\n",
      "hmmer-3.3/easel/esl_alloc.md\n",
      "hmmer-3.3/easel/easel.h\n",
      "hmmer-3.3/easel/esl_alloc.h\n",
      "hmmer-3.3/easel/interface_gsl.c\n",
      "hmmer-3.3/easel/esl_msaweight.md\n",
      "hmmer-3.3/easel/esl_mpi.h\n",
      "hmmer-3.3/easel/esl_rand64.c\n",
      "hmmer-3.3/easel/esl_stopwatch.h\n",
      "hmmer-3.3/easel/esl_neon.h\n",
      "hmmer-3.3/easel/esl_hmm.h\n",
      "hmmer-3.3/easel/esl_quicksort.c\n",
      "hmmer-3.3/easel/esl_rootfinder.tex\n",
      "hmmer-3.3/easel/m4/\n",
      "hmmer-3.3/easel/esl_sq.tex\n",
      "hmmer-3.3/easel/esl_sqio.c\n",
      "hmmer-3.3/easel/esl_randomseq.h\n",
      "hmmer-3.3/easel/esl_fileparser.c\n",
      "hmmer-3.3/easel/esl_json.c\n",
      "hmmer-3.3/easel/makeTAGS.sh\n",
      "hmmer-3.3/easel/esl_mixgev.h\n",
      "hmmer-3.3/easel/esl_stretchexp.tex\n",
      "hmmer-3.3/easel/esl_stack.tex\n",
      "hmmer-3.3/easel/devkit/\n",
      "hmmer-3.3/easel/esl_dirichlet.tex\n",
      "hmmer-3.3/easel/esl_msaweight.tex\n",
      "hmmer-3.3/easel/esl_sqio_ncbi.h\n",
      "hmmer-3.3/easel/esl_rootfinder.c\n",
      "hmmer-3.3/easel/esl_ratematrix.h\n",
      "hmmer-3.3/easel/esl_msaweight.h\n",
      "hmmer-3.3/easel/esl_msacluster.tex\n",
      "hmmer-3.3/easel/esl_sqio_ascii.c\n",
      "hmmer-3.3/easel/esl_vectorops.h\n",
      "hmmer-3.3/easel/esl_arr2.h\n",
      "hmmer-3.3/easel/esl_fileparser.tex\n",
      "hmmer-3.3/easel/esl_hyperexp.c\n",
      "hmmer-3.3/easel/esl_exponential.tex\n",
      "hmmer-3.3/easel/esl_workqueue.c\n",
      "hmmer-3.3/easel/esl_paml.tex\n",
      "hmmer-3.3/easel/decoy_config.h.in\n",
      "hmmer-3.3/easel/esl_msafile_phylip.h\n",
      "hmmer-3.3/easel/esl_tree.h\n",
      "hmmer-3.3/easel/esl_mixdchlet.tex\n",
      "hmmer-3.3/easel/esl_msa.h\n",
      "hmmer-3.3/easel/esl_msafile2.c\n",
      "hmmer-3.3/easel/esl_stats.tex\n",
      "hmmer-3.3/easel/esl_buffer.c\n",
      "hmmer-3.3/easel/easel.tex\n",
      "hmmer-3.3/easel/Makefile.in\n",
      "hmmer-3.3/easel/.travis.yml\n",
      "hmmer-3.3/easel/esl_avx.h\n",
      "hmmer-3.3/easel/esl_mixdchlet.c\n",
      "hmmer-3.3/easel/esl_dmatrix.c\n",
      "hmmer-3.3/easel/esl_msafile_clustal.h\n",
      "hmmer-3.3/easel/esl_gumbel.c\n",
      "hmmer-3.3/easel/esl_msafile.c\n",
      "hmmer-3.3/easel/esl_msafile_afa.h\n",
      "hmmer-3.3/easel/esl_stack.h\n",
      "hmmer-3.3/easel/esl_huffman.h\n",
      "hmmer-3.3/easel/esl_cpu.c\n",
      "hmmer-3.3/easel/interface_lapack.c\n",
      "hmmer-3.3/easel/esl_normal.c\n",
      "hmmer-3.3/easel/esl_varint.h\n",
      "hmmer-3.3/easel/esl_minimizer.tex\n",
      "hmmer-3.3/easel/esl_wuss.c\n",
      "hmmer-3.3/easel/esl_gamma.h\n",
      "hmmer-3.3/easel/esl_bitfield.c\n",
      "hmmer-3.3/easel/esl_msafile_psiblast.h\n",
      "hmmer-3.3/easel/esl_msafile.md\n",
      "hmmer-3.3/easel/esl_dsqdata.h\n",
      "hmmer-3.3/easel/esl_msa_testfiles/\n",
      "hmmer-3.3/easel/esl_threads.c\n",
      "hmmer-3.3/easel/esl_minimizer.h\n",
      "hmmer-3.3/easel/esl_msacluster.c\n",
      "hmmer-3.3/easel/esl_msafile_selex.c\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/a2m/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/psiblast/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/afa/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/clustal/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylips/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.6\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.8\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.7\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.5\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.4\n",
      "hmmer-3.3/easel/esl_msa_testfiles/stockholm/stockholm.bad.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylips/phylips.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylips/phylips.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/clustal/clustal.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/clustal/clustal.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/afa/afa.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/afa/afa.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/afa/afa.good.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/psiblast/psiblast.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/psiblast/psiblast.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/a2m/a2m.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/a2m/a2m.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.12\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.13\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.14\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.9\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.7\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.6\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.good.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.8\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.11\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.10\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.good.4\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.4\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.good.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/selex.bad.5\n",
      "hmmer-3.3/easel/esl_msa_testfiles/selex/00MANIFEST\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.11\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.10\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.good.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.4\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.3\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.good.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.2\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.5\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.ambig.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.9\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.7\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.6\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.1\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.bad.8\n",
      "hmmer-3.3/easel/esl_msa_testfiles/phylip/phylip.good.1\n",
      "hmmer-3.3/easel/devkit/techtree.py\n",
      "hmmer-3.3/easel/devkit/ctags-fix\n",
      "hmmer-3.3/easel/devkit/cexcerpt.man\n",
      "hmmer-3.3/easel/devkit/autodoc\n",
      "hmmer-3.3/easel/devkit/man2optlist\n",
      "hmmer-3.3/easel/devkit/autodoc.md\n",
      "hmmer-3.3/easel/devkit/sedition-pp\n",
      "hmmer-3.3/easel/devkit/00README\n",
      "hmmer-3.3/easel/devkit/rmanprocess.pl\n",
      "hmmer-3.3/easel/devkit/rmanprocess.py\n",
      "hmmer-3.3/easel/devkit/cexcerpt\n",
      "hmmer-3.3/easel/devkit/c2optlist\n",
      "hmmer-3.3/easel/devkit/sqc\n",
      "hmmer-3.3/easel/devkit/autodoc.py\n",
      "hmmer-3.3/easel/devkit/sedition\n",
      "hmmer-3.3/easel/m4/esl_avx.m4\n",
      "hmmer-3.3/easel/m4/esl_pic_flags.m4\n",
      "hmmer-3.3/easel/m4/esl_vmx.m4\n",
      "hmmer-3.3/easel/m4/ax_compiler_vendor.m4\n",
      "hmmer-3.3/easel/m4/ax_mpi.m4\n",
      "hmmer-3.3/easel/m4/esl_sse.m4\n",
      "hmmer-3.3/easel/m4/ax_check_compile_flag.m4\n",
      "hmmer-3.3/easel/m4/esl_avx512.m4\n",
      "hmmer-3.3/easel/m4/esl_sse4.m4\n",
      "hmmer-3.3/easel/m4/esl_cuda.m4\n",
      "hmmer-3.3/easel/m4/ax_check_gnu_make.m4\n",
      "hmmer-3.3/easel/m4/ax_gcc_func_attribute.m4\n",
      "hmmer-3.3/easel/m4/ax_pthread.m4\n",
      "hmmer-3.3/easel/m4/esl_neon.m4\n",
      "hmmer-3.3/easel/release-notes/RELEASE-0.46.md\n",
      "hmmer-3.3/easel/testsuite/i2-ncbi-indices.pl\n",
      "hmmer-3.3/easel/testsuite/trna-ssdraw.ps\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hmmer-3.3/easel/testsuite/e2.sh\n",
      "hmmer-3.3/easel/testsuite/parse_test_results.py\n",
      "hmmer-3.3/easel/testsuite/i3-blank-gf.pl\n",
      "hmmer-3.3/easel/testsuite/i1-degen-residues.pl\n",
      "hmmer-3.3/easel/testsuite/valgrind_report.pl\n",
      "hmmer-3.3/easel/testsuite/testsuite.sqc\n",
      "hmmer-3.3/easel/testsuite/trna-5.stk\n",
      "hmmer-3.3/easel/testsuite/Makefile.in\n",
      "hmmer-3.3/easel/testsuite/coverage_report.pl\n",
      "hmmer-3.3/easel/demotic/fasta2profmark\n",
      "hmmer-3.3/easel/demotic/blast2tbl\n",
      "hmmer-3.3/easel/demotic/demotic_infernal_tab.pm\n",
      "hmmer-3.3/easel/demotic/regress/\n",
      "hmmer-3.3/easel/demotic/demotic_blast.pm\n",
      "hmmer-3.3/easel/demotic/00README\n",
      "hmmer-3.3/easel/demotic/test.pl\n",
      "hmmer-3.3/easel/demotic/demotic_fasta.pm\n",
      "hmmer-3.3/easel/demotic/hmmer2profmark\n",
      "hmmer-3.3/easel/demotic/infernal_tab2gff.pl\n",
      "hmmer-3.3/easel/demotic/demotic_hmmer.pm\n",
      "hmmer-3.3/easel/demotic/examples/\n",
      "hmmer-3.3/easel/demotic/blast2profmark\n",
      "hmmer-3.3/easel/demotic/h22tbl\n",
      "hmmer-3.3/easel/demotic/fasta2tbl\n",
      "hmmer-3.3/easel/demotic/h22profmark\n",
      "hmmer-3.3/easel/demotic/demotic_h2.pm\n",
      "hmmer-3.3/easel/demotic/hmmer2tbl\n",
      "hmmer-3.3/easel/demotic/examples/example-single.fa.pin\n",
      "hmmer-3.3/easel/demotic/examples/example-single-psiquery.fa\n",
      "hmmer-3.3/easel/demotic/examples/example-single.pbl\n",
      "hmmer-3.3/easel/demotic/examples/example.hmm\n",
      "hmmer-3.3/easel/demotic/examples/example.hmmsearch.out\n",
      "hmmer-3.3/easel/demotic/examples/example-single.fa.psq\n",
      "hmmer-3.3/easel/demotic/examples/example.fa\n",
      "hmmer-3.3/easel/demotic/examples/example.ssearch.out\n",
      "hmmer-3.3/easel/demotic/examples/example-single.fa\n",
      "hmmer-3.3/easel/demotic/examples/example.sto\n",
      "hmmer-3.3/easel/demotic/examples/example.hmm2\n",
      "hmmer-3.3/easel/demotic/examples/example-single.asnt\n",
      "hmmer-3.3/easel/demotic/examples/example.psiblast.out\n",
      "hmmer-3.3/easel/demotic/examples/example-single.sto\n",
      "hmmer-3.3/easel/demotic/examples/example.wu-blastp.out\n",
      "hmmer-3.3/easel/demotic/examples/example.fasta.out\n",
      "hmmer-3.3/easel/demotic/examples/example.h2.out\n",
      "hmmer-3.3/easel/demotic/examples/example.phmmer.out\n",
      "hmmer-3.3/easel/demotic/examples/example-single.fa.phr\n",
      "hmmer-3.3/easel/demotic/examples/example.ncbi-blastp.out\n",
      "hmmer-3.3/easel/demotic/regress/ssearch-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/hmmsearch-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/psiblast-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/h2-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/ncbi-blastp-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/wu-blastp-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/ncbi-blastp-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/wu-blastp-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/fasta-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/h2-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/fasta-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/phmmer-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/phmmer-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/ssearch-tbl.out\n",
      "hmmer-3.3/easel/demotic/regress/hmmsearch-profmark.out\n",
      "hmmer-3.3/easel/demotic/regress/psiblast-tbl.out\n",
      "hmmer-3.3/easel/formats/fasta.bad.1\n",
      "hmmer-3.3/easel/formats/wag.dat\n",
      "hmmer-3.3/easel/formats/BLOSUM62\n",
      "hmmer-3.3/easel/formats/fasta\n",
      "hmmer-3.3/easel/formats/fasta.2\n",
      "hmmer-3.3/easel/formats/fasta.bad.2\n",
      "hmmer-3.3/easel/formats/genbank.2\n",
      "hmmer-3.3/easel/formats/fasta.bad.3\n",
      "hmmer-3.3/easel/formats/embl\n",
      "hmmer-3.3/easel/formats/uniprot\n",
      "hmmer-3.3/easel/formats/genbank\n",
      "hmmer-3.3/easel/formats/stockholm.1\n",
      "hmmer-3.3/easel/formats/genbank.bad.1\n",
      "hmmer-3.3/easel/formats/fasta.odd.1\n",
      "hmmer-3.3/easel/documentation/copyright.tex\n",
      "hmmer-3.3/easel/documentation/intro.tex\n",
      "hmmer-3.3/easel/documentation/format_stockholm.tex\n",
      "hmmer-3.3/easel/documentation/typography.tex\n",
      "hmmer-3.3/easel/documentation/format_ncbi.tex\n",
      "hmmer-3.3/easel/documentation/macros.tex\n",
      "hmmer-3.3/easel/documentation/doctricks.tex\n",
      "hmmer-3.3/easel/documentation/install.tex\n",
      "hmmer-3.3/easel/documentation/main.tex\n",
      "hmmer-3.3/easel/documentation/format_wuss.tex\n",
      "hmmer-3.3/easel/documentation/format_a2m.tex\n",
      "hmmer-3.3/easel/documentation/figures/\n",
      "hmmer-3.3/easel/documentation/codestyle.tex\n",
      "hmmer-3.3/easel/documentation/titlepage.tex\n",
      "hmmer-3.3/easel/documentation/Makefile.in\n",
      "hmmer-3.3/easel/documentation/installation.tex\n",
      "hmmer-3.3/easel/documentation/codestyle.md\n",
      "hmmer-3.3/easel/documentation/statistics.tex\n",
      "hmmer-3.3/easel/documentation/figures/easel_techtree.png\n",
      "hmmer-3.3/easel/documentation/figures/easel_techtree.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-bsu-alignment.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rna_elements.pdf\n",
      "hmmer-3.3/easel/documentation/figures/evd_basic.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rna_elements.ai\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-bsu-alignment.ai\n",
      "hmmer-3.3/easel/documentation/figures/evd_location.pdf\n",
      "hmmer-3.3/easel/documentation/figures/gev_density.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-ecoli.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-ecoli.eps\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-ecoli.ai\n",
      "hmmer-3.3/easel/documentation/figures/rnaseP-bsu-alignment.eps\n",
      "hmmer-3.3/easel/documentation/figures/evd_scale.pdf\n",
      "hmmer-3.3/easel/documentation/figures/easel_techtree.ai\n",
      "hmmer-3.3/easel/documentation/figures/gev_logdensity.pdf\n",
      "hmmer-3.3/easel/documentation/figures/rna_elements.eps\n",
      "hmmer-3.3/easel/miniapps/esl-seqstat.c\n",
      "hmmer-3.3/easel/miniapps/esl-mixdchlet.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimask.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-alimanip.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimanip.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-compalign.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-mask.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-shuffle.itest.py\n",
      "hmmer-3.3/easel/miniapps/esl-alimap.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-alistat.c\n",
      "hmmer-3.3/easel/miniapps/easel-downsample.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-translate.c\n",
      "hmmer-3.3/easel/miniapps/esl-shuffle.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-alistat.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-mask.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-reformat.c\n",
      "hmmer-3.3/easel/miniapps/esl-histplot.c\n",
      "hmmer-3.3/easel/miniapps/esl-alirev.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimerge.itest.pl\n",
      "hmmer-3.3/easel/miniapps/easel.c\n",
      "hmmer-3.3/easel/miniapps/esl-reformat.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-seqrange.c\n",
      "hmmer-3.3/easel/miniapps/esl-seqstat.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-shuffle.c\n",
      "hmmer-3.3/easel/miniapps/esl-translate.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-seqrange.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-compstruct.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimap.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimap.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-translate.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-afetch.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-alimerge.c\n",
      "hmmer-3.3/easel/miniapps/esl-compalign.c\n",
      "hmmer-3.3/easel/miniapps/esl-construct.c\n",
      "hmmer-3.3/easel/miniapps/esl-ssdraw.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-alimask.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimanip.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-mask.c\n",
      "hmmer-3.3/easel/miniapps/esl-weight.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-alipid.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimask.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-weight.c\n",
      "hmmer-3.3/easel/miniapps/cmd_filter.md\n",
      "hmmer-3.3/easel/miniapps/cmd_alistat.c\n",
      "hmmer-3.3/easel/miniapps/esl-ssdraw.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-seqrange.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-histplot.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-construct.itest.pl\n",
      "hmmer-3.3/easel/miniapps/esl-afetch.c\n",
      "hmmer-3.3/easel/miniapps/esl-alimerge.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-ssdraw.c\n",
      "hmmer-3.3/easel/miniapps/esl-selectn.c\n",
      "hmmer-3.3/easel/miniapps/cmd_filter.c\n",
      "hmmer-3.3/easel/miniapps/esl-alipid.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-selectn.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-construct.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-alirev.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-mixdchlet.man.in\n",
      "hmmer-3.3/easel/miniapps/cmd_downsample.c\n",
      "hmmer-3.3/easel/miniapps/esl-afetch.itest.pl\n",
      "hmmer-3.3/easel/miniapps/Makefile.in\n",
      "hmmer-3.3/easel/miniapps/esl-sfetch.c\n",
      "hmmer-3.3/easel/miniapps/esl-alistat.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-compalign.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-sfetch.man.in\n",
      "hmmer-3.3/easel/miniapps/esl-compstruct.man.in\n",
      "hmmer-3.3/profmark/x-h2-ls\n",
      "hmmer-3.3/profmark/x-psiblast+\n",
      "hmmer-3.3/profmark/x-h2-fs\n",
      "hmmer-3.3/profmark/x-single-ssearch\n",
      "hmmer-3.3/profmark/pmark-master.pl\n",
      "hmmer-3.3/profmark/x-iterate-psiblast\n",
      "hmmer-3.3/profmark/x-fps-fasta\n",
      "hmmer-3.3/profmark/00README\n",
      "hmmer-3.3/profmark/x-fps-ssearch\n",
      "hmmer-3.3/profmark/x-phmmer-consensus\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hmmer-3.3/profmark/rocplot.pl\n",
      "hmmer-3.3/profmark/x-single-phmmer\n",
      "hmmer-3.3/profmark/x-fps-ncbiblast+\n",
      "hmmer-3.3/profmark/x-hmmsearch\n",
      "hmmer-3.3/profmark/x-sam\n",
      "hmmer-3.3/profmark/x-iterate-jackhmmer\n",
      "hmmer-3.3/profmark/rocplot.c\n",
      "hmmer-3.3/profmark/pmark.param\n",
      "hmmer-3.3/profmark/x-fps-wublast\n",
      "hmmer-3.3/profmark/create-profmark.c\n",
      "hmmer-3.3/profmark/x-hmmsearch-max\n",
      "hmmer-3.3/profmark/x-fps-ncbiblast\n",
      "hmmer-3.3/profmark/x-fps-phmmer\n",
      "hmmer-3.3/profmark/Makefile.in\n",
      "hmmer-3.3/profmark/x-single-ncbiblast\n",
      "hmmer-3.3/profmark/x-psiblast\n",
      "hmmer-3.3/testsuite/20aa-alitest.afa\n",
      "hmmer-3.3/testsuite/20aa-alitest.phr\n",
      "hmmer-3.3/testsuite/RRM_1.sto\n",
      "hmmer-3.3/testsuite/3box-alitest.nhr\n",
      "hmmer-3.3/testsuite/20aa-alitest.sto\n",
      "hmmer-3.3/testsuite/i15-hmmconvert.pl\n",
      "hmmer-3.3/testsuite/i19-hmmpgmd-ga.pl\n",
      "hmmer-3.3/testsuite/2OG-FeII_Oxy_3.hmm\n",
      "hmmer-3.3/testsuite/20aa-alitest.psi\n",
      "hmmer-3.3/testsuite/RRM_1.hmm\n",
      "hmmer-3.3/testsuite/h3.pm\n",
      "hmmer-3.3/testsuite/3box-alitest.fa\n",
      "hmmer-3.3/testsuite/3box-alitest.nsi\n",
      "hmmer-3.3/testsuite/XYPPX.hmm\n",
      "hmmer-3.3/testsuite/i8-nonresidues.pl\n",
      "hmmer-3.3/testsuite/iss159-nhmmer-overlap.py\n",
      "hmmer-3.3/testsuite/i22-hmmpgmd-shard-ga.pl\n",
      "hmmer-3.3/testsuite/i7-hmmbuild-fragments.pl\n",
      "hmmer-3.3/testsuite/i11-hmmalign-mapali.pl\n",
      "hmmer-3.3/testsuite/i21-rewind.pl\n",
      "hmmer-3.3/testsuite/20aa-alitest.pog\n",
      "hmmer-3.3/testsuite/i12-delete-corruption.pl\n",
      "hmmer-3.3/testsuite/XYPPX.sto\n",
      "hmmer-3.3/testsuite/3box-alitest.nog\n",
      "hmmer-3.3/testsuite/i2-search-variation.sh\n",
      "hmmer-3.3/testsuite/ecori.hmm\n",
      "hmmer-3.3/testsuite/i18-nhmmer-generic.pl\n",
      "hmmer-3.3/testsuite/ecori.fa\n",
      "hmmer-3.3/testsuite/20aa.hmm\n",
      "hmmer-3.3/testsuite/parse_test_results.py\n",
      "hmmer-3.3/testsuite/3box.hmm\n",
      "hmmer-3.3/testsuite/M1.sto\n",
      "hmmer-3.3/testsuite/i14-hmmemit-consensus.pl\n",
      "hmmer-3.3/testsuite/2OG-FeII_Oxy_3-nt.hmm\n",
      "hmmer-3.3/testsuite/PSE.sto\n",
      "hmmer-3.3/testsuite/minifam\n",
      "hmmer-3.3/testsuite/20aa-alitest.psd\n",
      "hmmer-3.3/testsuite/PSE.hmm\n",
      "hmmer-3.3/testsuite/rndseq400-10.fa\n",
      "hmmer-3.3/testsuite/M1.hmm\n",
      "hmmer-3.3/testsuite/20aa-alitest.psq\n",
      "hmmer-3.3/testsuite/testsuite.sqc\n",
      "hmmer-3.3/testsuite/3box-alitest.nsd\n",
      "hmmer-3.3/testsuite/3box.sto\n",
      "hmmer-3.3/testsuite/20aa.sto\n",
      "hmmer-3.3/testsuite/ecori.sto\n",
      "hmmer-3.3/testsuite/i3-seqsearch-variation.sh\n",
      "hmmer-3.3/testsuite/3box-alitest.nsq\n",
      "hmmer-3.3/testsuite/LuxC.hmm\n",
      "hmmer-3.3/testsuite/i23-bad-fasta.sh\n",
      "hmmer-3.3/testsuite/SMC_N.sto\n",
      "hmmer-3.3/testsuite/Caudal_act.sto\n",
      "hmmer-3.3/testsuite/PAX8_HUMAN\n",
      "hmmer-3.3/testsuite/i5-hmmbuild-naming.pl\n",
      "hmmer-3.3/testsuite/20aa-alitest.fa\n",
      "hmmer-3.3/testsuite/i17-stdin.pl\n",
      "hmmer-3.3/testsuite/Patched.hmm\n",
      "hmmer-3.3/testsuite/i1-build-variation.sh\n",
      "hmmer-3.3/testsuite/Patched.sto\n",
      "hmmer-3.3/testsuite/Makefile.in\n",
      "hmmer-3.3/testsuite/i4-zerolength-seqs.sh\n",
      "hmmer-3.3/testsuite/Caudal_act.hmm\n",
      "hmmer-3.3/testsuite/i10-duplicate-names.pl\n",
      "hmmer-3.3/testsuite/3box-alitest.nin\n",
      "hmmer-3.3/testsuite/i6-hmmalign-mapali.pl\n",
      "hmmer-3.3/testsuite/SMC_N.hmm\n",
      "hmmer-3.3/testsuite/i20-fmindex-core.pl\n",
      "hmmer-3.3/testsuite/i9-optional-annotation.pl\n",
      "hmmer-3.3/testsuite/LuxC.sto\n",
      "hmmer-3.3/testsuite/20aa-alitest.pin\n",
      "hmmer-3.3/testsuite/i16-build-allins.pl\n",
      "hmmer-3.3/testsuite/i13-msa-integrity.pl\n",
      "hmmer-3.3/test-speed/x-psiblast+\n",
      "hmmer-3.3/test-speed/x-ssearch\n",
      "hmmer-3.3/test-speed/00README\n",
      "hmmer-3.3/test-speed/x-hmmer2\n",
      "hmmer-3.3/test-speed/x-ncbiblast+\n",
      "hmmer-3.3/test-speed/speed-master.pl\n",
      "hmmer-3.3/test-speed/speedA-list.pl\n",
      "hmmer-3.3/test-speed/x-hmmsearch\n",
      "hmmer-3.3/test-speed/x-sam\n",
      "hmmer-3.3/test-speed/component-benchmark.pl\n",
      "hmmer-3.3/test-speed/x-ncbiblast\n",
      "hmmer-3.3/test-speed/x-wublast\n",
      "hmmer-3.3/test-speed/x-fasta\n",
      "hmmer-3.3/test-speed/x-psiblast\n",
      "hmmer-3.3/libdivsufsort/AUTHORS\n",
      "hmmer-3.3/libdivsufsort/divsufsort.h.in\n",
      "hmmer-3.3/libdivsufsort/divsufsort.c\n",
      "hmmer-3.3/libdivsufsort/README\n",
      "hmmer-3.3/libdivsufsort/COPYING\n",
      "hmmer-3.3/libdivsufsort/VERSION\n",
      "hmmer-3.3/libdivsufsort/Makefile.in\n",
      "hmmer-3.3/tutorial/MADE1.sto\n",
      "hmmer-3.3/tutorial/Pkinase.hmm\n",
      "hmmer-3.3/tutorial/Pkinase.sto\n",
      "hmmer-3.3/tutorial/dna_target.fa\n",
      "hmmer-3.3/tutorial/MADE1.hmm\n",
      "hmmer-3.3/tutorial/globins45.fa\n",
      "hmmer-3.3/tutorial/fn3.hmm\n",
      "hmmer-3.3/tutorial/fn3.sto\n",
      "hmmer-3.3/tutorial/globins4.sto\n",
      "hmmer-3.3/tutorial/globins4.hmm\n",
      "hmmer-3.3/tutorial/7LESS_DROME\n",
      "hmmer-3.3/tutorial/HBB_HUMAN\n",
      "hmmer-3.3/documentation/man/\n",
      "hmmer-3.3/documentation/userguide/\n",
      "hmmer-3.3/documentation/Makefile.in\n",
      "hmmer-3.3/documentation/userguide/titlepage_daemon.tex.in\n",
      "hmmer-3.3/documentation/userguide/tufte-common-local.tex\n",
      "hmmer-3.3/documentation/userguide/tutorial.tex\n",
      "hmmer-3.3/documentation/userguide/daemon.tex\n",
      "hmmer-3.3/documentation/userguide/macros.tex\n",
      "hmmer-3.3/documentation/userguide/hmmc.tex\n",
      "hmmer-3.3/documentation/userguide/install.tex\n",
      "hmmer-3.3/documentation/userguide/inclusions/\n",
      "hmmer-3.3/documentation/userguide/tufte-book.cls\n",
      "hmmer-3.3/documentation/userguide/main.tex\n",
      "hmmer-3.3/documentation/userguide/tabular.tex\n",
      "hmmer-3.3/documentation/userguide/copyright.tex.in\n",
      "hmmer-3.3/documentation/userguide/formats.tex\n",
      "hmmer-3.3/documentation/userguide/introduction.tex\n",
      "hmmer-3.3/documentation/userguide/tufte-common.def\n",
      "hmmer-3.3/documentation/userguide/unsrtnat-brief.bst\n",
      "hmmer-3.3/documentation/userguide/titlepage.tex.in\n",
      "hmmer-3.3/documentation/userguide/glossary.tex\n",
      "hmmer-3.3/documentation/userguide/pipeline.tex\n",
      "hmmer-3.3/documentation/userguide/distilled.bib\n",
      "hmmer-3.3/documentation/userguide/Makefile.in\n",
      "hmmer-3.3/documentation/userguide/hmmbuild.tex\n",
      "hmmer-3.3/documentation/userguide/ack.tex\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmscan-noargs.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-globins.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmalign-globins.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmstat-minifam.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmbuild-noargs.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hit-serialize.ai\n",
      "hmmer-3.3/documentation/userguide/inclusions/domain-serialize.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/inclusions.def\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out4\n",
      "hmmer-3.3/documentation/userguide/inclusions/daemon.ai\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out3\n",
      "hmmer-3.3/documentation/userguide/inclusions/nhmmer-made1.out3\n",
      "hmmer-3.3/documentation/userguide/inclusions/alidisplay-serialize.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/nhmmer-made1.out4\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmpress-minifam.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/daemon.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/nhmmer-made1.out5\n",
      "hmmer-3.3/documentation/userguide/inclusions/nhmmer-made1.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/daemon-results.ai\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out5\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmscan-minifam-sevenless.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/domain-serialize.ai\n",
      "hmmer-3.3/documentation/userguide/inclusions/daemon-results.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmbuild-globins.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmbuild-globins.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/nhmmer-made1.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-fn3-sevenless.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-fn3-uniprot.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmpgmd-search-stats-serialize.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmalign-globins.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-fn3-sevenless.out4\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-fn3-sevenless.out3\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmc2_2.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out7\n",
      "hmmer-3.3/documentation/userguide/inclusions/hit-serialize.pdf\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmscan-minifam-sevenless.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-globins.out2\n",
      "hmmer-3.3/documentation/userguide/inclusions/alidisplay-serialize.ai\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-globins.out3\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmscan-minifam-sevenless.out3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hmmer-3.3/documentation/userguide/inclusions/hmmbuild-made1.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/jackhmmer-hbb-uniprot.out6\n",
      "hmmer-3.3/documentation/userguide/inclusions/globins4.sto\n",
      "hmmer-3.3/documentation/userguide/inclusions/Makefile.in\n",
      "hmmer-3.3/documentation/userguide/inclusions/gen-inclusions.py\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmsearch-fn3-sevenless.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmc2.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/sevenless_domains.out\n",
      "hmmer-3.3/documentation/userguide/inclusions/hmmpgmd-search-stats-serialize.ai\n",
      "hmmer-3.3/documentation/man/nhmmscan.man.in\n",
      "hmmer-3.3/documentation/man/phmmer.man.in\n",
      "hmmer-3.3/documentation/man/makehmmerdb.man.in\n",
      "hmmer-3.3/documentation/man/hmmconvert.man.in\n",
      "hmmer-3.3/documentation/man/hmmalign.man.in\n",
      "hmmer-3.3/documentation/man/hmmpgmd.man.in\n",
      "hmmer-3.3/documentation/man/hmmstat.man.in\n",
      "hmmer-3.3/documentation/man/hmmc2.man.in\n",
      "hmmer-3.3/documentation/man/hmmpgmd_shard.man.in\n",
      "hmmer-3.3/documentation/man/alimask.man.in\n",
      "hmmer-3.3/documentation/man/hmmpress.man.in\n",
      "hmmer-3.3/documentation/man/hmmfetch.man.in\n",
      "hmmer-3.3/documentation/man/nhmmer.man.in\n",
      "hmmer-3.3/documentation/man/hmmsearch.man.in\n",
      "hmmer-3.3/documentation/man/hmmer.man.in\n",
      "hmmer-3.3/documentation/man/hmmsim.man.in\n",
      "hmmer-3.3/documentation/man/hmmlogo.man.in\n",
      "hmmer-3.3/documentation/man/jackhmmer.man.in\n",
      "hmmer-3.3/documentation/man/hmmbuild.man.in\n",
      "hmmer-3.3/documentation/man/hmmemit.man.in\n",
      "hmmer-3.3/documentation/man/Makefile.in\n",
      "hmmer-3.3/documentation/man/hmmscan.man.in\n"
     ]
    }
   ],
   "source": [
    "wget http://eddylab.org/software/hmmer/hmmer-3.3.tar.gz\n",
    "tar -xvzf hmmer-3.3.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Compile `hmmer`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "configure: Configuring HMMER3 for your system.\n",
      "checking build system type... x86_64-pc-linux-gnu\n",
      "checking host system type... x86_64-pc-linux-gnu\n",
      "checking for gcc... gcc\n",
      "checking whether the C compiler works... yes\n",
      "checking for C compiler default output file name... a.out\n",
      "checking for suffix of executables... \n",
      "checking whether we are cross compiling... no\n",
      "checking for suffix of object files... o\n",
      "checking whether we are using the GNU C compiler... yes\n",
      "checking whether gcc accepts -g... yes\n",
      "checking for gcc option to accept ISO C89... none needed\n",
      "checking for gcc option to accept ISO C99... none needed\n",
      "checking for gcc option to accept ISO Standard C... (cached) none needed\n",
      "checking how to run the C preprocessor... gcc -E\n",
      "checking for a BSD-compatible install... /bin/install -c\n",
      "checking for ranlib... ranlib\n",
      "checking for ar... /bin/ar\n",
      "checking whether ln -s works... yes\n",
      "checking for a sed that does not truncate output... /bin/sed\n",
      "checking for grep that handles long lines and -e... /bin/grep\n",
      "checking for egrep... /bin/grep -E\n",
      "checking whether gcc is Clang... no\n",
      "checking whether pthreads work with -pthread... yes\n",
      "checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE\n",
      "checking whether more special flags are required for pthreads... no\n",
      "checking for PTHREAD_PRIO_INHERIT... yes\n",
      "checking whether gcc can compile our Altivec/VMX code... no\n",
      "checking whether gcc can compile our SSE code... yes\n",
      "configure: Activating Intel/AMD SSE vector DP implementation\n",
      "checking whether _MM_SET_FLUSH_ZERO_MODE is supported... yes\n",
      "checking compiler support --start-group... yes\n",
      "checking for gzip... /bin/gzip\n",
      "checking for python3... yes\n",
      "checking for library containing clock_gettime... none required\n",
      "checking for ANSI C header files... yes\n",
      "checking for sys/types.h... yes\n",
      "checking for sys/stat.h... yes\n",
      "checking for stdlib.h... yes\n",
      "checking for string.h... yes\n",
      "checking for memory.h... yes\n",
      "checking for strings.h... yes\n",
      "checking for inttypes.h... yes\n",
      "checking for stdint.h... yes\n",
      "checking for unistd.h... yes\n",
      "checking endian.h usability... yes\n",
      "checking endian.h presence... yes\n",
      "checking for endian.h... yes\n",
      "checking for inttypes.h... (cached) yes\n",
      "checking for stdint.h... (cached) yes\n",
      "checking for unistd.h... (cached) yes\n",
      "checking for sys/types.h... (cached) yes\n",
      "checking netinet/in.h usability... yes\n",
      "checking netinet/in.h presence... yes\n",
      "checking for netinet/in.h... yes\n",
      "checking sys/param.h usability... yes\n",
      "checking sys/param.h presence... yes\n",
      "checking for sys/param.h... yes\n",
      "checking for sys/sysctl.h... yes\n",
      "checking for uint8_t... yes\n",
      "checking for uint16_t... yes\n",
      "checking for uint32_t... yes\n",
      "checking for uint64_t... yes\n",
      "checking for off_t... yes\n",
      "checking whether byte ordering is bigendian... no\n",
      "checking for __attribute__((noreturn))... yes\n",
      "checking for __attribute__((format))... yes\n",
      "checking for mkstemp... yes\n",
      "checking for popen... yes\n",
      "checking for putenv... yes\n",
      "checking for strcasecmp... yes\n",
      "checking for strsep... yes\n",
      "checking for times... yes\n",
      "checking for getpid... yes\n",
      "checking for sysctl... yes\n",
      "checking for sysconf... yes\n",
      "checking for getcwd... yes\n",
      "checking for chmod... yes\n",
      "checking for stat... yes\n",
      "checking for fstat... yes\n",
      "checking for erfc... no\n",
      "checking for library containing ntohs... none required\n",
      "checking for library containing ntohl... none required\n",
      "checking for library containing htons... none required\n",
      "checking for library containing htonl... none required\n",
      "checking for library containing socket... none required\n",
      "checking for library containing inet_pton... none required\n",
      "checking for _LARGEFILE_SOURCE value needed for large files... no\n",
      "checking for special C compiler options needed for large files... no\n",
      "checking for _FILE_OFFSET_BITS value needed for large files... no\n",
      "configure: creating ./config.status\n",
      "config.status: creating Makefile\n",
      "config.status: creating src/Makefile\n",
      "config.status: creating testsuite/Makefile\n",
      "config.status: creating profmark/Makefile\n",
      "config.status: creating src/impl_sse/Makefile\n",
      "config.status: creating documentation/Makefile\n",
      "config.status: creating documentation/man/Makefile\n",
      "config.status: creating documentation/userguide/Makefile\n",
      "config.status: creating documentation/userguide/inclusions/Makefile\n",
      "config.status: creating libdivsufsort/Makefile\n",
      "config.status: creating easel/Makefile\n",
      "config.status: creating easel/miniapps/Makefile\n",
      "config.status: creating easel/testsuite/Makefile\n",
      "config.status: creating easel/documentation/Makefile\n",
      "config.status: creating documentation/userguide/titlepage.tex\n",
      "config.status: creating documentation/userguide/titlepage_daemon.tex\n",
      "config.status: creating documentation/userguide/copyright.tex\n",
      "config.status: creating documentation/man/alimask.man\n",
      "config.status: creating documentation/man/hmmalign.man\n",
      "config.status: creating documentation/man/hmmbuild.man\n",
      "config.status: creating documentation/man/hmmc2.man\n",
      "config.status: creating documentation/man/hmmconvert.man\n",
      "config.status: creating documentation/man/hmmemit.man\n",
      "config.status: creating documentation/man/hmmer.man\n",
      "config.status: creating documentation/man/hmmfetch.man\n",
      "config.status: creating documentation/man/hmmlogo.man\n",
      "config.status: creating documentation/man/hmmpgmd.man\n",
      "config.status: creating documentation/man/hmmpgmd_shard.man\n",
      "config.status: creating documentation/man/hmmpress.man\n",
      "config.status: creating documentation/man/hmmscan.man\n",
      "config.status: creating documentation/man/hmmsearch.man\n",
      "config.status: creating documentation/man/hmmsim.man\n",
      "config.status: creating documentation/man/hmmstat.man\n",
      "config.status: creating documentation/man/jackhmmer.man\n",
      "config.status: creating documentation/man/makehmmerdb.man\n",
      "config.status: creating documentation/man/nhmmer.man\n",
      "config.status: creating documentation/man/nhmmscan.man\n",
      "config.status: creating documentation/man/phmmer.man\n",
      "config.status: creating easel/miniapps/esl-afetch.man\n",
      "config.status: creating easel/miniapps/esl-alimanip.man\n",
      "config.status: creating easel/miniapps/esl-alimap.man\n",
      "config.status: creating easel/miniapps/esl-alimask.man\n",
      "config.status: creating easel/miniapps/esl-alimerge.man\n",
      "config.status: creating easel/miniapps/esl-alipid.man\n",
      "config.status: creating easel/miniapps/esl-alirev.man\n",
      "config.status: creating easel/miniapps/esl-alistat.man\n",
      "config.status: creating easel/miniapps/esl-compalign.man\n",
      "config.status: creating easel/miniapps/esl-compstruct.man\n",
      "config.status: creating easel/miniapps/esl-construct.man\n",
      "config.status: creating easel/miniapps/esl-histplot.man\n",
      "config.status: creating easel/miniapps/esl-mask.man\n",
      "config.status: creating easel/miniapps/esl-mixdchlet.man\n",
      "config.status: creating easel/miniapps/esl-reformat.man\n",
      "config.status: creating easel/miniapps/esl-selectn.man\n",
      "config.status: creating easel/miniapps/esl-seqrange.man\n",
      "config.status: creating easel/miniapps/esl-seqstat.man\n",
      "config.status: creating easel/miniapps/esl-sfetch.man\n",
      "config.status: creating easel/miniapps/esl-shuffle.man\n",
      "config.status: creating easel/miniapps/esl-ssdraw.man\n",
      "config.status: creating easel/miniapps/esl-translate.man\n",
      "config.status: creating easel/miniapps/esl-weight.man\n",
      "config.status: creating easel/decoy_config.h\n",
      "config.status: creating src/p7_config.h\n",
      "config.status: creating easel/esl_config.h\n",
      "config.status: creating libdivsufsort/divsufsort.h\n",
      "config.status: linking src/impl_sse to src/impl\n",
      "\n",
      "\n",
      "HMMER configuration:\n",
      "   compiler:             gcc -O3   -pthread \n",
      "   host:                 x86_64-pc-linux-gnu\n",
      "   linker:               \n",
      "   libraries:              \n",
      "   DP implementation:    sse\n",
      "\n",
      "Now do 'make'  to build HMMER, and optionally:\n",
      "       'make check'  to run self tests,\n",
      "       'make install'  to install programs and man pages,\n",
      "       '(cd easel; make install)'  to install Easel tools.\n",
      "\n",
      "     SUBDIR easel\n",
      "     CC easel.o\n",
      "     CC esl_alloc.o\n",
      "     CC esl_alphabet.o\n",
      "     CC esl_arr2.o\n",
      "     CC esl_arr3.o\n",
      "     CC esl_bitfield.o\n",
      "     CC esl_buffer.o\n",
      "     CC esl_cluster.o\n",
      "     CC esl_composition.o\n",
      "     CC esl_cpu.o\n",
      "     CC esl_dirichlet.o\n",
      "     CC esl_distance.o\n",
      "     CC esl_dmatrix.o\n",
      "     CC esl_dsqdata.o\n",
      "     CC esl_exponential.o\n",
      "     CC esl_fileparser.o\n",
      "     CC esl_gamma.o\n",
      "     CC esl_gencode.o\n",
      "     CC esl_getopts.o\n",
      "     CC esl_gev.o\n",
      "     CC esl_graph.o\n",
      "     CC esl_gumbel.o\n",
      "     CC esl_heap.o\n",
      "     CC esl_histogram.o\n",
      "     CC esl_hmm.o\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     CC esl_huffman.o\n",
      "     CC esl_hyperexp.o\n",
      "     CC esl_json.o\n",
      "     CC esl_keyhash.o\n",
      "     CC esl_matrixops.o\n",
      "     CC esl_mem.o\n",
      "     CC esl_minimizer.o\n",
      "     CC esl_mixdchlet.o\n",
      "     CC esl_mixgev.o\n",
      "     CC esl_mpi.o\n",
      "     CC esl_msa.o\n",
      "     CC esl_msacluster.o\n",
      "     CC esl_msafile.o\n",
      "     CC esl_msafile2.o\n",
      "     CC esl_msafile_a2m.o\n",
      "     CC esl_msafile_afa.o\n",
      "     CC esl_msafile_clustal.o\n",
      "     CC esl_msafile_phylip.o\n",
      "     CC esl_msafile_psiblast.o\n",
      "     CC esl_msafile_selex.o\n",
      "     CC esl_msafile_stockholm.o\n",
      "     CC esl_msashuffle.o\n",
      "     CC esl_msaweight.o\n",
      "     CC esl_normal.o\n",
      "     CC esl_paml.o\n",
      "     CC esl_quicksort.o\n",
      "     CC esl_random.o\n",
      "     CC esl_rand64.o\n",
      "     CC esl_randomseq.o\n",
      "     CC esl_ratematrix.o\n",
      "     CC esl_recorder.o\n",
      "     CC esl_red_black.o\n",
      "     CC esl_regexp.o\n",
      "     CC esl_rootfinder.o\n",
      "     CC esl_scorematrix.o\n",
      "     CC esl_sq.o\n",
      "     CC esl_sqio.o\n",
      "     CC esl_sqio_ascii.o\n",
      "     CC esl_sqio_ncbi.o\n",
      "     CC esl_ssi.o\n",
      "     CC esl_stack.o\n",
      "     CC esl_stats.o\n",
      "     CC esl_stopwatch.o\n",
      "     CC esl_stretchexp.o\n",
      "     CC esl_subcmd.o\n",
      "     CC esl_threads.o\n",
      "     CC esl_tree.o\n",
      "     CC esl_varint.o\n",
      "     CC esl_vectorops.o\n",
      "     CC esl_weibull.o\n",
      "     CC esl_workqueue.o\n",
      "     CC esl_wuss.o\n",
      "     CC esl_sse.o\n",
      "     CC esl_avx.o\n",
      "     CC esl_avx512.o\n",
      "     CC esl_neon.o\n",
      "     CC esl_vmx.o\n",
      "     AR libeasel.a\n",
      "     SUBDIR miniapps\n",
      "     GEN esl-afetch\n",
      "     GEN esl-alimanip\n",
      "     GEN esl-alimap\n",
      "     GEN esl-alimask\n",
      "     GEN esl-alimerge\n",
      "     GEN esl-alipid\n",
      "     GEN esl-alirev\n",
      "     GEN esl-alistat\n",
      "     GEN esl-compalign\n",
      "     GEN esl-compstruct\n",
      "     GEN esl-construct\n",
      "     GEN esl-histplot\n",
      "     GEN esl-mask\n",
      "     GEN esl-mixdchlet\n",
      "     GEN esl-reformat\n",
      "     GEN esl-selectn\n",
      "     GEN esl-seqrange\n",
      "     GEN esl-seqstat\n",
      "     GEN esl-sfetch\n",
      "     GEN esl-shuffle\n",
      "     GEN esl-ssdraw\n",
      "     GEN esl-translate\n",
      "     GEN esl-weight\n",
      "     CC cmd_alistat.o\n",
      "     CC cmd_downsample.o\n",
      "     CC cmd_filter.o\n",
      "     GEN easel\n",
      "     SUBDIR libdivsufsort\n",
      "     CC divsufsort.o\n",
      "     AR libdivsufsort.a\n",
      "     SUBDIR src\n",
      "     CC alimask.o\n",
      "     CC build.o\n",
      "     CC cachedb.o\n",
      "     CC cachedb_shard.o\n",
      "     CC emit.o\n",
      "     CC errors.o\n",
      "     CC evalues.o\n",
      "     CC eweight.o\n",
      "     CC generic_decoding.o\n",
      "     CC generic_fwdback.o\n",
      "     CC generic_fwdback_chk.o\n",
      "     CC generic_fwdback_banded.o\n",
      "     CC generic_null2.o\n",
      "     CC generic_msv.o\n",
      "     CC generic_optacc.o\n",
      "     CC generic_stotrace.o\n",
      "     CC generic_viterbi.o\n",
      "     CC generic_vtrace.o\n",
      "     CC h2_io.o\n",
      "     CC heatmap.o\n",
      "     CC hmmlogo.o\n",
      "     CC hmmdmstr.o\n",
      "     CC hmmdmstr_shard.o\n",
      "     CC hmmd_search_status.o\n",
      "     CC hmmdwrkr.o\n",
      "     CC hmmdwrkr_shard.o\n",
      "     CC hmmdutils.o\n",
      "     CC hmmer.o\n",
      "     CC logsum.o\n",
      "     CC modelconfig.o\n",
      "     CC modelstats.o\n",
      "     CC mpisupport.o\n",
      "     CC seqmodel.o\n",
      "     CC tracealign.o\n",
      "     CC p7_alidisplay.o\n",
      "     CC p7_bg.o\n",
      "     CC p7_builder.o\n",
      "     CC p7_domain.o\n",
      "     CC p7_domaindef.o\n",
      "     CC p7_gbands.o\n",
      "     CC p7_gmx.o\n",
      "     CC p7_gmxb.o\n",
      "     CC p7_gmxchk.o\n",
      "     CC p7_hit.o\n",
      "     CC p7_hmm.o\n",
      "     CC p7_hmmcache.o\n",
      "     CC p7_hmmd_search_stats.o\n",
      "     CC p7_hmmfile.o\n",
      "     CC p7_hmmwindow.o\n",
      "     CC p7_pipeline.o\n",
      "     CC p7_prior.o\n",
      "     CC p7_profile.o\n",
      "     CC p7_spensemble.o\n",
      "     CC p7_tophits.o\n",
      "     CC p7_trace.o\n",
      "     CC p7_scoredata.o\n",
      "     CC hmmpgmd2msa.o\n",
      "     CC fm_alphabet.o\n",
      "     CC fm_general.o\n",
      "     CC fm_sse.o\n",
      "     CC fm_ssv.o\n",
      "     AR libhmmer-src.stamp\n",
      "     SUBDIR impl_sse\n",
      "     CC decoding.o\n",
      "     CC fwdback.o\n",
      "     CC io.o\n",
      "     CC ssvfilter.o\n",
      "     CC msvfilter.o\n",
      "     CC null2.o\n",
      "     CC optacc.o\n",
      "     CC stotrace.o\n",
      "     CC vitfilter.o\n",
      "     CC p7_omx.o\n",
      "     CC p7_oprofile.o\n",
      "     CC mpi.o\n",
      "     AR libhmmer-impl.stamp\n",
      "     GEN alimask\n",
      "     CC hmmalign.o\n",
      "     GEN hmmalign\n",
      "     CC hmmbuild.o\n",
      "     GEN hmmbuild\n",
      "     CC hmmconvert.o\n",
      "     GEN hmmconvert\n",
      "     CC hmmemit.o\n",
      "     GEN hmmemit\n",
      "     CC hmmfetch.o\n",
      "     GEN hmmfetch\n",
      "     GEN hmmlogo\n",
      "     CC hmmpgmd.o\n",
      "     GEN hmmpgmd\n",
      "     CC hmmpgmd_shard.o\n",
      "     GEN hmmpgmd_shard\n",
      "     CC hmmpress.o\n",
      "     GEN hmmpress\n",
      "     CC hmmscan.o\n",
      "     GEN hmmscan\n",
      "     CC hmmsearch.o\n",
      "     GEN hmmsearch\n",
      "     CC hmmsim.o\n",
      "     GEN hmmsim\n",
      "     CC hmmstat.o\n",
      "     GEN hmmstat\n",
      "     CC jackhmmer.o\n",
      "     GEN jackhmmer\n",
      "     CC phmmer.o\n",
      "     GEN phmmer\n",
      "     CC nhmmer.o\n",
      "     GEN nhmmer\n",
      "     CC nhmmscan.o\n",
      "     GEN nhmmscan\n",
      "     CC makehmmerdb.o\n",
      "     GEN makehmmerdb\n",
      "     CC hmmc2.o\n",
      "     GEN hmmc2\n",
      "     CC hmmerfm-exactmatch.o\n",
      "     GEN hmmerfm-exactmatch\n",
      "     SUBDIR profmark\n",
      "     CC create-profmark.o\n",
      "     GEN create-profmark\n",
      "     CC rocplot.o\n",
      "     GEN rocplot\n"
     ]
    }
   ],
   "source": [
    "cd hmmer-3.3\n",
    "./configure\n",
    "make"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Put `hmmer` in path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/user/home/secook2/info302/info302_assignment/blast-2.2.26/bin\n",
      "/data/user/home/secook2/info302/info302_assignment/blast-2.2.26/bin/hmmer-3.3/src/hmmbuild\n"
     ]
    }
   ],
   "source": [
    "current_dir=`pwd`\n",
    "cd src\n",
    "export PATH=$PATH:`pwd`\n",
    "cd $current_dir\n",
    "cd ..\n",
    "pwd\n",
    "which hmmbuild"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run muscle to create the alignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "MUSCLE v3.8.31 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "p53_all 8 seqs, max length 393, avg  length 384\n",
      "00:00:00    14 MB(-6%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00    14 MB(-6%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00    18 MB(-8%)  Iter   1  100.00%  Align node       \n",
      "00:00:00    18 MB(-8%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00    18 MB(-8%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00    18 MB(-8%)  Iter   3  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   4  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   5  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   6  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   7  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   7  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   8  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter   9  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  10  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  11  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  12  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  13  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  14  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  15  100.00%  Refine biparts\n",
      "00:00:00    18 MB(-8%)  Iter  16  100.00%  Refine biparts\n"
     ]
    }
   ],
   "source": [
    "./muscle -in p53_all.fas -out p53.aln"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create the hmm."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.3 (Nov 2019); http://hmmer.org/\n",
      "# Copyright (C) 2019 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             p53.aln\n",
      "# output HMM file:                  p53.hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     p53                      8   410   387     0.61  0.592 \n",
      "\n",
      "# CPU time: 0.30u 0.00s 00:00:00.30 Elapsed: 00:00:00.35\n"
     ]
    }
   ],
   "source": [
    "hmmbuild --informat afa p53.hmm p53.aln"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Search the hmm against swissprot database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "engine": "bash",
    "eval": false
   },
   "outputs": [],
   "source": [
    "hmmsearch -o hits.txt p53.hmm uniprot_sprot.fasta.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check the result."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmsearch :: search profile(s) against a sequence database\n",
      "# HMMER 3.3 (Nov 2019); http://hmmer.org/\n",
      "# Copyright (C) 2019 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# query HMM file:                  p53.hmm\n",
      "# target sequence database:        uniprot_sprot.fasta.gz\n",
      "# output directed to file:         hits.txt\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "cat: write error: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "cat hits.txt | head"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "eval,language,cache,engine,-all",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
