{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes on FASTQ/BAM/SAM files\n",
    "\n",
    "### BAM files\n",
    "\n",
    "- compressed version of SAM\n",
    "\n",
    "- bam file is a binary so wer cant read and contains unmapped reads, multiple alignments on the same read and suppulentary reads  \n",
    "\n",
    "### SAM Files\n",
    "\n",
    "- sam is tab delimted for storing read alignments to a reference genome \n",
    "\n",
    "- One record (a single DNA fragment alignment) per line describing alignment between fragmetns and reference \n",
    "\n",
    "- 11 fixed colums + optional key:type:value tuples\n",
    "\n",
    "- 12 colums listed in SAM file:\n",
    "\n",
    "    > Query Name \n",
    "    \n",
    "    > Bitwise FLAG\n",
    "    \n",
    "    > reference sequence name \n",
    "    \n",
    "    > position 1-bated left most position of clipped alignment \n",
    "    \n",
    "    >mapping quality (needs to be greater than 20 for a good score)\n",
    "    \n",
    "    >extended CIGAR string (if the nucleotide is a match/instertion/deletion)\n",
    "    \n",
    "    >Mate reference NaMe\n",
    "    \n",
    "    >1-based leftmost mate position \n",
    "    \n",
    "    >infered insert size\n",
    "    \n",
    "    >query sequence on the dame strand as the reference \n",
    "    \n",
    "    >query quality \n",
    "    \n",
    "    >optional fields \n",
    "    \n",
    "\n",
    "- Flags are the number of bits (12 bit long strings ) representing the flags present of ) 0= false , 1=true\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Quality control of data \n",
    "\n",
    "- done in 2 stages \n",
    "\n",
    "    1) just after fastq files initial quality check to test if fastq files are good. most common one is fastqc and generates report on the file. \n",
    "    \n",
    "       > the x axis represents the length of the sequence, all files will be of identical length \n",
    "       \n",
    "       > for each postition we can get the phred score of each and create a box plot and then graph the medians of the values agaist the y axis\n",
    "       \n",
    "       > as lenght increases quality decreases \n",
    "       \n",
    "- Base calling errors \n",
    "\n",
    "    > phasing noise \n",
    "    \n",
    "    >signal delay \n",
    "    \n",
    "    > mixed cluster\n",
    "    \n",
    "- Auto QC test done after quality control "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
